src/CTA.Rules.Update/CodeReplacer.cs

```diff
@@ -83,10 +89,50 @@ public CodeReplacer(List<SourceFileBuildResult> sourceFileBuildResults, ProjectC
                 {
                     LogHelper.LogError(new FilePortingException(Constants.Project, new Exception("Error adding project to actions collection")));
                 }
+
+                if (_projectConfiguration.ProjectType == ProjectType.WCFConfigBasedService || _projectConfiguration.ProjectType == ProjectType.WCFCodeBasedService)
+                {
+                    RunWCFChanges();
```

Comment:
if there is an exception are we catching and logging as metrics ?

Reply:
+1. You are updating multiple files in this function. You should catch the result and make sure it's logged if anything goes wrong.